Search out of memory: CPU2GPU1 with index flat

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
when do search in the collection (50m entities, segment_row_limit=2048, index=flat), and topk=[1,10,100,1000] nq=200, the error `out of memory: CPU2GPU1` occurred. during search in node poseidon, the gpu-memory usage is up to 6000MiB (total is 8119 MiB)
```
10:53:08  2020-11-04:02:53:07,237 DEBUG    [client.py:293] Row count: 50000000 in collection: <sift_50m_2048_128_l2>
10:53:08  2020-11-04:02:53:07,238 INFO     [k8s_runner.py:377] 50000000
10:53:08  2020-11-04:02:53:07,241 INFO     [k8s_runner.py:379] {'index_type': 'flat', 'index_param': {'nlist': 16384}}
10:55:30  2020-11-04:02:55:29,491 INFO     [client.py:46] Milvus preload_collection run in 142.25s
10:55:30  2020-11-04:02:55:29,491 INFO     [k8s_runner.py:381] Start warm up query
10:55:30  2020-11-04:02:55:29,760 INFO     [runner.py:196] Start query, query params: top-k: 1, nq: 1, actually length of vectors: 1
10:55:30  2020-11-04:02:55:29,760 INFO     [runner.py:198] Start run query, run 1 of 2
10:55:32  2020-11-04:02:55:32,272 INFO     [runner.py:198] Start run query, run 2 of 2
10:55:34  2020-11-04:02:55:33,945 INFO     [runner.py:204] Min query time: 1.67
10:55:34  2020-11-04:02:55:33,959 INFO     [k8s_runner.py:383] End warm up query
10:55:34  2020-11-04:02:55:33,959 INFO     [k8s_runner.py:385] Search param: {"nprobe": 8}
10:55:34  2020-11-04:02:55:34,88 INFO     [runner.py:196] Start query, query params: top-k: 1, nq: 200, actually length of vectors: 200
10:55:34  2020-11-04:02:55:34,88 INFO     [runner.py:198] Start run query, run 1 of 2
10:55:41  2020-11-04:02:55:40,516 INFO     [runner.py:198] Start run query, run 2 of 2
10:55:46  2020-11-04:02:55:45,800 INFO     [runner.py:204] Min query time: 5.28
10:55:46  2020-11-04:02:55:45,800 INFO     [runner.py:196] Start query, query params: top-k: 10, nq: 200, actually length of vectors: 200
10:55:46  2020-11-04:02:55:45,801 INFO     [runner.py:198] Start run query, run 1 of 2
10:55:50  2020-11-04:02:55:50,80 ERROR    [client.py:102] sift_50m_2048_128_l2
10:55:50  2020-11-04:02:55:50,80 ERROR    [client.py:103] out of memory: CPU2GPU1 : Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2150400000 bytes (error 2 out of memory)
10:55:50  2020-11-04:02:55:50,113 ERROR    [client.py:104] (Status(code=0, message='Success!'), 'OK')
10:55:50  2020-11-04:02:55:50,132 DEBUG    [client.py:289] (Status(code=0, message='Success!'), 50000000)
10:55:50  2020-11-04:02:55:50,137 DEBUG    [client.py:293] Row count: 50000000 in collection: <sift_50m_2048_128_l2>
10:55:50  2020-11-04:02:55:50,137 ERROR    [client.py:105] 50000000
10:55:50  2020-11-04:02:55:50,137 ERROR    [main.py:70] Status not ok
10:55:50  2020-11-04:02:55:50,140 ERROR    [main.py:71] Traceback (most recent call last):
10:55:50    File "main.py", line 68, in queue_worker
10:55:50      runner.run(run_type, collection)
10:55:50    File "/home/jenkins/agent/workspace/milvus-benchmark-0.10.4/milvus_benchmark/k8s_runner.py", line 386, in run
10:55:50      res = self.do_query(milvus_instance, collection_name, top_ks, nqs, run_count, search_param)
10:55:50    File "/home/jenkins/agent/workspace/milvus-benchmark-0.10.4/milvus_benchmark/runner.py", line 200, in do_query
10:55:50      query_res = milvus.query(vectors, top_k, search_param=search_param)
10:55:50    File "/home/jenkins/agent/workspace/milvus-benchmark-0.10.4/milvus_benchmark/client.py", line 263, in query
10:55:50      self.check_status(status)
10:55:50    File "/home/jenkins/agent/workspace/milvus-benchmark-0.10.4/milvus_benchmark/client.py", line 106, in check_status
10:55:50      raise Exception("Status not ok")
10:55:50  Exception: Status not ok
```
the log path:
`/test/milvus/db_data_8/sift_50m_2048_128_l2_flat/logs/milvus-20-11-04-10:52-debug.log`
`/test/milvus/db_data_8/sift_50m_2048_128_l2_flat/logs/milvus-20-11-04-10:52-error.log`

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [ ] Docker/cpu
- [x] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)
 -0.10.4

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
search_performance:
  collections:
    -
      server:
        db_config.primary_path: /test/milvus/db_data_8/sift_50m_2048_128_l2_flat
        cache_config.cpu_cache_capacity: 64GB
        engine_config.use_blas_threshold: 0
        engine_config.gpu_search_threshold: 200
        gpu_resource_config.enable: true
        gpu_resource_config.cache_capacity: 6GB
        gpu_resource_config.search_resources:
          - gpu0
          - gpu1
        gpu_resource_config.build_index_resources:
          - gpu0
          - gpu1
        wal_enable: true
      collection_name: sift_50m_2048_128_l2
      run_count: 2
      top_ks: [1, 10, 100, 1000]
      nqs: [200]
      search_params:
        -
          nprobe: 8
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@ Please mark all change in change log and use the issue from GitHub
 -   \#3906 Change DeleteTask state when it is loaded to avoid server crash.
 -   \#4012 Milvus hangs when continually creating and dropping partitions.
 -   \#4075 Improve performance for create large amount of partitions
+-   \#4174 Search out of memory: CPU2GPU1 with index flat
 
 ## Feature
 -   \#3773 Support IVF_PQ to run on FPGA.

--- a/core/src/scheduler/TaskTable.cpp
+++ b/core/src/scheduler/TaskTable.cpp
@@ -165,7 +165,7 @@ TaskTable::PickToLoad(uint64_t limit) {
         } else if (table_[index]->state == TaskTableItemState::LOADED) {
             cross = true;
             ++loaded_count;
-            if (loaded_count > 2)
+            if (loaded_count >= 1)
                 return std::vector<uint64_t>();
         } else if (table_[index]->state == TaskTableItemState::START) {
             auto task = table_[index]->task;

--- a/core/unittest/scheduler/test_resource.cpp
+++ b/core/unittest/scheduler/test_resource.cpp
@@ -24,7 +24,7 @@
 namespace milvus {
 namespace scheduler {
 
-constexpr uint64_t max_once_load = 2;
+constexpr uint64_t max_once_load = 1;
 
 /************ ResourceBaseTest ************/
 class ResourceBaseTest : public testing::Test {

